lib/matplotlib/path.py

```diff
@@ -1042,3 +1203,248 @@ def get_path_collection_extents(
+        # equivalent by symmetry, but keeps math simpler
+        phi = np.pi - phi
+    # if there's no corner (i.e. the path just ends, as in the "sides" of the
+    # caret marker (and other non-fillable markers), we still need to compute
+    # how much the "cap" extends past the endpoint of the path
+    if theta is None:
+        # for "butt" caps we can compute how far the
+        # outside edge of the markeredge stroke extends outside of the bounding
+        # box of its path using the law of sines: $\sin(\phi)/(w/2) =
+        # \sin(\pi/2 - \phi)/l$ for $w$ the `markeredgewidth`, $\phi$ the
+        # incidence angle of the line, then $l$ is the length along the outer
+        # edge of the stroke that extends beyond the bouding box. We can
+        # translate this to a distance perpendicular to the bounding box E(w,
+        # \phi) = l \sin(\phi)$, for $l$ as above.
+        if capstyle == 'butt':
+            return (width/2) * np.cos(phi)
+        # "round" caps are hemispherical, so regardless of angle
+        elif capstyle == 'round':
+            return width/2
+        # finally, projecting caps are just bevel caps with an extra
+        # width/2 distance along the direction of the line, so "butt" plus some
+        # extra
+        elif capstyle == 'projecting':
+            return (width/2) * np.cos(phi) + (width/2)*np.sin(phi)
+        else:
+            raise ValueError(f"Unknown capstyle: {capstyle}.")
+    # the two "same as straight line" cases are NaN limits in the miter formula
+    elif (np.isclose(theta, 0) and np.isclose(phi, 0)
+          or np.isclose(theta, np.pi) and np.isclose(phi, np.pi/2)):
+        return width/2
+    # to calculate the offset for _joinstyle == 'miter', imagine aligning the
+    # corner so that one line comes in along the negative x-axis, and another
+    # from above, making an angle $\theta$ with the negative x-axis. The tip of
+    # the new corner created by the markeredge stroke will be at the point
+    # where the two outer edge of the markeredge stroke intersect.  in the
+    # orientation described above, the outer edge of the stroke aligned with
+    # the x-axis will obviously have equation $y = -w/2$ where $w$ is the
+    # markeredgewidth. WLOG, the stroke coming in from above at an angle
+    # $\theta$ from the negative x-axis will have equation
+    # $$-(\tan(\theta) x + \frac{w}{2\cos(\theta)}.$$
+    # the intersection of these two lines is at $y = w/2$, and we can solve for
+    # $x = \cot(\theta) (\frac{w}{2} + \frac{w}{2\cos(\theta)})$.
+    # this puts the "edge" tip a distance $M = (w/2)\csc(\theta/2)$
+    # from the tip of the corner itself, on the line defined by the bisector of
+    # the corner angle. So the extra padding required is $M\sin(\phi)$, where
+    # $\phi$ is the incidence angle of the corner's bisector. Notice that in
+    # the limit ($\phi = \theta/2$) where the "corner" is flush with the bbox,
+    # this correctly simplifies to just $w/2$.
```

Comment:
Somewhere around this point, I think the reference for angles changed from the negative x-axis to the positive y-axis without stating it, or else the sine doesn't make sense to me.